ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+        }
+    }
+
+    /**
+     * check if the id of the line match with the id of a tie line in ucte
+     *
+     * @param line is a tie line ?
+     * @return true if the parameter is a tie line
+     */
+    boolean isUcteTieLineId(Line line) {
+
+        MergedXnode mergedXnode = line.getExtension(MergedXnode.class);
+
+        return mergedXnode != null
+                && mergedXnode.getLine1Name() != null
+                && mergedXnode.getLine2Name() != null
+                && mergedXnode.getLine1Name().length() >= 19
+                && mergedXnode.getLine2Name().length() >= 19
+                && isUcteNodeId(mergedXnode.getLine1Name().substring(0, 8))
+                && isUcteNodeId(mergedXnode.getLine1Name().substring(9, 17))
+                && isUcteNodeId(mergedXnode.getLine2Name().substring(0, 8))
+                && isUcteNodeId(mergedXnode.getLine2Name().substring(9, 17));
+    }
+
+    private void setSwitchCurrentLimit(UcteLine ucteLine, Switch sw) {
+        if (sw.getProperties().containsKey(UcteImporter.CURRENT_LIMIT_PROPERTY_KEY)) {
+            try {
+                ucteLine.setCurrentLimit(Integer.parseInt(sw.getProperties().getProperty(UcteImporter.CURRENT_LIMIT_PROPERTY_KEY)));
+            } catch (NumberFormatException exception) {
+                ucteLine.setCurrentLimit(UcteImporter.DEFAULT_MAX_CURRENT);
+                LOGGER.warn("Switch {}: No current limit, set value to {}", sw.getId(), UcteImporter.DEFAULT_MAX_CURRENT);
+            }
+        } else {
+            ucteLine.setCurrentLimit(UcteImporter.DEFAULT_MAX_CURRENT);
+            LOGGER.warn("Switch {}: No current limit, set value to {}", sw.getId(), UcteImporter.DEFAULT_MAX_CURRENT);
+        }
+    }
+
+    String getGeographicalNameProperty(Line line) {
+        if (line.getProperties().containsKey(UcteImporter.GEOGRAPHICAL_NAME_PROPERTY_KEY)) {
+            return line.getProperties().getProperty(UcteImporter.GEOGRAPHICAL_NAME_PROPERTY_KEY);
+        }
+        return "";
+    }
+
+    UctePowerPlantType energySourceToUctePowerPlantType(EnergySource energySource) {
+        if (EnergySource.HYDRO == energySource) {
+            return UctePowerPlantType.H;
+        } else if (EnergySource.NUCLEAR == energySource) {
+            return UctePowerPlantType.N;
+        } else if (EnergySource.THERMAL == energySource) {
+            return UctePowerPlantType.C;
+        } else if (EnergySource.WIND == energySource) {
+            return UctePowerPlantType.W;
+        } else {
+            return UctePowerPlantType.F;
+        }
+    }
+
+    /**
+     * "increment" the string given in parameter by 1 : "aaaa" gives "aaab", "aaaz" gives "aaaA" etc...
+     * It goes from a -> z -> A -> Z -> 0 -> 9 and it's used to generate a random geographicalName.
+     * @param toIncrement the string to increment
+     * @return a string "incremented" by 1
+     */
+    String incrementGeneratedGeographicalName(String toIncrement) {
```

Comment:
Should this method be private?

Reply:
This method is tested separetaley, reason why it is private package I think. But, related to following comments, it shows that there is a need for dedicated mechanism/strategy for these IDs generation that should be public (and well tested ...)